### ISSUE TITLE
[Feature Request] Prebuilt `*.deb` packages for TT software

### ISSUE THREAD

**Issue by marty1885** [tenstorrent/tt-metal#7915]
**Is your feature request related to a problem? Please describe.**

As of now installing the base TT stack is difficult and some users are unable to do so without much effort. I'll help if TT could provide a prebuilt Metalium DEB package. Along with other dependencies (tt-kmd and tt-smi) also in DEB format. This enables most users to install and start developing TT software on their Ubuntu machine easily.

Furthermore I'll be nice to have a [PPA](https://launchpad.net/ubuntu/+ppas) so people can automatically update when TT released a new version of the software.

Imagine Nvidia requires everyone to build CUDA and nvidia-smi from source for every computer - that won't work right? 

**Describe the solution you'd like**

1. Provide prebuilt DEB packages to Ubuntu (and maybe Debian) users
2. Create a PPA so users can install and update automatically

**Describe alternatives you've considered**

**Additional context**

**Comment by jliangTT** [comment]
Related to packaging, and triage to infra team to comment. Setting it as P2 as it is likely not yet on the roadmap.

**Comment by TT-billteng** [comment]
Hi @marty1885, we are looking into adding conda support as our go-to method for distribution. Have you used it before?

Conda doesn't require `sudo`, and the support burden is less than `deb` packages as it works across all flavors of Linux + windows/MacOS (some more work needed).

**Comment by marty1885** [comment]
@TT-billteng Yes I am familiar with conda. I think there are some important points I need to bring up. The following are non issues with `deb` packages (I think you already know, just in case).

* A `deb` package for tt-kmd is still needed. Still need a way to get the OS to handle DKMS update every now and then when the kernel is updated
  * Conda Can't install drivers
  * With deb packages. You can simply specify the dependency of each Metalium/tt-smi build to depend on a specific version of tt-kmd. With conda.. They are separate and needs explicit management,
* You need _really_ good CMake integration and guides.
  * Most CMake is not written to work with conda
  * People will struggle if there is no "how to" easily available.
  
I am more concerned about the 2nd one. But if you can get it working nicely, cool!

**Comment by TT-billteng** [comment]
> @TT-billteng Yes I am familiar with conda. I think there are some important points I need to bring up. The following are non issues with `deb` packages (I think you already know, just in case).
> 
> * A `deb` package for tt-kmd is still needed. Still need a way to get the OS to handle DKMS update every now and then when the kernel is updated

Yes, however Metalium is also a consumer of tt-kmd so this should be an issue filed on kmd repo :D

> * You need _really_ good CMake integration and guides.
>   
>   * Most CMake is not written to work with conda
>   * People will struggle if there is no "how to" easily available.
> 
> I am more concerned about the 2nd one. But if you can get it working nicely, cool!

Yes, but this is the flow I'm most familiar with and if we do it properly, it's less work for us overall

**Comment by tt-rkim** [comment]
We're slowly chugging away at this.
